### PR TITLE
Typescript integration

### PR DESCRIPTION
**Pull request by tomaskikutis**
SDESK-3226

### CODE REVIEW

File: scripts/tests.js [hunk 1]
```diff
@@ -9,6 +9,6 @@ import Adapter from 'enzyme-adapter-react-16';
 
 Enzyme.configure({adapter: new Adapter()});
 
-var testsContext = require.context('.', true, /.spec.[t|j]sx?$/);
+var testsContext = require.context('.', true, /.spec.(js|ts|tsx)$/);
```
Comment (pablopunk) [inline scripts/tests.js]: This means we won't use `.jsx` any more right?

Comment (tomaskikutis) [reply scripts/tests.js]: Yes, we'll no longer use `.jsx` - the extension, but will continue using `jsx` syntax in `.tsx` files.

**Comment by petrjasek** [comment]
would it be possible to avoid changes in existing files? it makes it harder to backport bugfixes to older versions

**Comment by tomaskikutis** [comment]
I could keep the existing files unchanged, but we wouldn't get any type checking then.

I left most of `*.js` files untouched and only changed `*.jsx` which allows them to be type-checked. All other files are just bundled for now and not type-checked. The plan is to merge this first, so we don't do too much at once.

If you look at the diff, it's mostly <5 lines per file, so it shouldn't be too hard to backport.

File: scripts/apps/search/components/actions-menu/Label.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import {closeActionsMenu} from '../../helpers';
 
-export default function Label(props) {
+export const Label:React.StatelessComponent<any> = (props) => {
```
Comment (pablopunk) [inline scripts/apps/search/components/actions-menu/Label.tsx]: not very important but can't we still `export default const Label...` ?

Comment (tomaskikutis) [reply scripts/apps/search/components/actions-menu/Label.tsx]: That might be important, seems like I removed default export. I'll fix that on Thursday. Please keep an eye on similar situations. Most of the changes should be simply adding types and not changing any logic. There are a few exceptions, but it should be obvious there.

Comment (pablopunk) [reply scripts/apps/search/components/actions-menu/Label.tsx]: Okay, also remember to take `import {Label}` back to `import Label`

File: scripts/core/ui/components/Form/SelectTagInput/index.tsx [hunk 3]
```diff
@@ -124,7 +128,7 @@ export class SelectTagInput extends React.Component {
                     <LineInput noMargin={true} invalid={invalid} required={required}>
                         <Label text={label} />
 
-                        <div className="sd-line-input__input" readOnly={readOnly}>
```
Comment (pablopunk) [inline scripts/core/ui/components/Form/SelectTagInput/index.tsx]: why are we removing `readOnly`?

File: scripts/core/ui/components/Toggle/index.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ import './style.scss';
  * @name Toggle
  * @description Toggle component used to togle check/uncheck status
  */
-export default function Toggle({value, onChange, readOnly, onFocus, className}) {
```
Comment (pablopunk) [inline scripts/core/ui/components/Toggle/index.tsx]: another default here

File: scripts/globals.d.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+declare const $ : any; // jquery
```
Comment (pablopunk) [inline scripts/globals.d.ts]: should we add `Window` to the list of globals?

File: scripts/core/ui/components/Form/SelectTagInput/index.tsx (hunk 3 above)
Comment (tomaskikutis) [reply scripts/core/ui/components/Form/SelectTagInput/index.tsx]: `readOnly` is an attribute for inputs and `div` is not an input. TypeScript flagged it an invalid property which it is.

**Comment by petrjasek** [comment]
what would be the idea regarding prop-types vs component interfaces?

File: webpack.config.js [hunk 6]
```diff
@@ -79,20 +79,15 @@ module.exports = function makeConfig(grunt) {
                 // ensure that react is loaded only once (3rd party apps can load more...)
                 react: path.resolve('./node_modules/react'),
             },
-            extensions: ['.js', '.jsx'],
+            extensions: ['.js', '.ts', '.tsx'],
```
Comment (petrjasek) [inline webpack.config.js]: I think we might need to keep jsx working for superdesk-planning

Comment (mdhaman) [reply webpack.config.js]: Guys be lenient allow both. Are we planning to migrate angular to 2+?

Comment (tomaskikutis) [reply webpack.config.js]: I forgot the apps are using the same config, will fix this. @mdhaman there are no plans for angular 2+ as far as I know.

**Comment by tomaskikutis** [comment]
> what would be the idea regarding prop-types vs component interfaces?

TypeScript checks are compile-time instead of run-time checks by prop-types, so I don't see any reasons to keep prop-types for long. I'm only keeping them now to save time and minimize changes in this PR.

**Comment by petrjasek** [comment]
I'm just wondering if it can check at compile time that react-redux connect mapping state/dispatch to props.

**Comment by tomaskikutis** [comment]
Looks like it's [possible](https://spin.atomicobject.com/2017/04/20/typesafe-container-components/).

File: package.json [hunk 7]
```diff
@@ -108,8 +108,6 @@
     "btoa": "^1.1.2",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
-    "eslint": "^5.3.0",
```
Comment (pablopunk) [inline package.json]: Shouldn't we still have `eslint` as a dependency as we are using it on `npm run lint`?

Comment (tomaskikutis) [reply package.json]: I've added a commit comment

> use eslint from superdesk-code-style

Eslint is already a dependency in `superdesk-code-style` repo, which is a dependency for this one, so eslint is already there. Not sure if this is a good practice, just leaving it as it were before this PR.

Comment (pablopunk) [reply package.json]: oh okay, I thought we had eslint before

**Pull request merged**